code/src/App.js

```diff
@@ -1,9 +1,115 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
+import moment from 'moment'
+import { API_URL } from './utils/urls'
+
 
 export const App = () => {
+
+
+  const [newThought, setNewThought] = useState([])
+  const [thoughts, setThoughts] = useState([])
+  const [characters, setCharacters] = useState([])
+
+
+  useEffect(() => {
+
+    fetch(API_URL)
+    .then((res) => res.json())
+    .then(data => setThoughts(data))
+
+  }, [] )
+
+
+  const onFormSubmit = (e) => {
+    e.preventDefault()
+    
+    fetch( API_URL, 
+        { method: 'POST', headers: { 'Content-Type': 'application/json',}, body: JSON.stringify({ message: newThought }) }
+      ).then((res) => res.json()).then((data) => setThoughts([data, ...thoughts])).then(setNewThought(''))
+
+      if (newThought.length < 4 ) {
+        alert('Too few characters')
+      }
+      else if ( newThought.length > 140 ) {
+        alert('Too many characters')
+      } 
+  }
+
+
+  const addLike = (thoughtId) => {
+
+    fetch(`https://happy-thoughts-technigo.herokuapp.com/thoughts/${thoughtId}/like`, 
+        { method: 'POST'}
+
+    ).then((res) => res.json()).then((data) => {
+
+      const updateLikes = thoughts.map((item) => {
+          if(item._id === data._id) {
+          item.hearts += 1
+          return item }
+          
+          else { return item }
+      })
+      setThoughts(updateLikes)
+    })
+  }
+
+
+  const characterStyling = (chars) => {
+
+    if (chars.length < 4  ||  chars.length > 140) {
+    return ('charactersNOTOK') }
+    else return ('charactersOK')
+  }
+
+
   return (
-    <div>
-      Find me in src/app.js!
+
+    <div className="thoughts-wrapper">
+
+    <form onSubmit={onFormSubmit}> 
+      <label htmlFor="newThought">What's making you happy right now?</label>
+      <input 
+      className={characterStyling(characters)} 
+      value={newThought} 
+      onInput={(e) => setCharacters(e.target.value)}
+      onChange={(e) => setNewThought(e.target.value)}
+      id={newThought}
```

Comment:
To limit the input value, we can use html minlength and maxlength attributes. Check out the link below for more info: [https://www.w3schools.com/tags/att_input_maxlength.asp](url)